mav_nonlinear_mpc/solver_made_from_cpp/nmpc_solver_setup.cpp

```diff
@@ -0,0 +1,124 @@
+    OnlineData external_forces3;
+
+    // Non-linear drag
+    IntermediateState dragacc1 =   sin(pitch)*linear_drag_coefficient1*thrust*velocity3
+                                 + cos(pitch)*cos(yaw)*linear_drag_coefficient1*thrust*velocity1
+                                 - cos(pitch)*linear_drag_coefficient1*sin(yaw)*thrust*velocity2;
+    IntermediateState dragacc2 =   (cos(roll)*sin(yaw) - cos(yaw)*sin(pitch)*sin(roll))*linear_drag_coefficient2*thrust*velocity1
+                                 - (cos(roll)*cos(yaw) + sin(pitch)*sin(roll)*sin(yaw))*linear_drag_coefficient2*thrust*velocity2
+                                 - cos(pitch)*linear_drag_coefficient2*sin(roll)*thrust*velocity3;
+
+    // Model equations:
+    DifferentialEquation f;
+
+    f << dot(velocity1)   == ((cos(roll)*cos(yaw)*sin(pitch) + sin(roll)*sin(yaw))*thrust - dragacc1 + external_forces1);
+    f << dot(velocity2)   == ((cos(roll)*sin(pitch)*sin(yaw) - cos(yaw)*sin(roll))*thrust - dragacc2 + external_forces2);
+    f << dot(velocity3)   == (-g + cos(pitch)*cos(roll)*thrust + external_forces3);
+    f << dot( roll )      == (roll_gain*roll_ref - roll)/roll_tau;
+    f << dot( pitch )     == (pitch_gain*pitch_ref - pitch)/pitch_tau;
+    f << dot( yaw )       == 0;
+    f << dot( position1 ) == velocity1;
+    f << dot( position2 ) == velocity2;
+    f << dot( position3 ) == velocity3;
+
+    // Reference functions and weighting matrices:
+    Function h;
+    h << position1 << position2 << position3;
+    h << velocity1 << velocity2 << velocity3;
+    h << roll      << pitch;
+    h << roll_ref  << pitch_ref << (cos(pitch)*cos(roll)*thrust - g);
+
+    Function hN;
+    hN << position1 << position2 << position3;
+    hN << velocity1 << velocity2 << velocity3;
+
+    // Provide defined weighting matrices:
+    BMatrix W  = eye<bool>(h.getDim());
+    BMatrix WN = eye<bool>(hN.getDim());
+
+    //Define OCP problem:
+    OCP ocp(0.0, N*Ts, N);
+
+    ocp.subjectTo(f);
+
+    ocp.minimizeLSQ(W, h);
+    ocp.minimizeLSQEndTerm(WN, hN);
+
+    // Dummy constraints, real ones set online
+    ocp.subjectTo(-45*PI/180 <= roll_ref  <= 45*PI/180);
+    ocp.subjectTo(-45*PI/180 <= pitch_ref <= 45*PI/180);
+    ocp.subjectTo(     g/2.0 <= thrust    <= g*1.5);
+
+    // Export the code:
+    OCPexport mpc( ocp );
+
+    mpc.set( HESSIAN_APPROXIMATION, GAUSS_NEWTON);
+    mpc.set( DISCRETIZATION_TYPE, MULTIPLE_SHOOTING);
+    mpc.set( SPARSE_QP_SOLUTION, FULL_CONDENSING_N2); //FULL_CONDENsinG_N2
+    mpc.set( INTEGRATOR_TYPE, INT_IRK_GL2);
+    //mpc.set( NUM_INTEGRATOR_STEPS, N);
+    mpc.set( QP_SOLVER, QP_QPOASES);
+    mpc.set( HOTSTART_QP, NO);
+    mpc.set( LEVENBERG_MARQUARDT, 1e-10);
+    mpc.set( LINEAR_ALGEBRA_SOLVER, GAUSS_LU);
+    mpc.set( IMPLICIT_INTEGRATOR_NUM_ITS, 2);
+    mpc.set( CG_USE_OPENMP, YES);
+    mpc.set( CG_HARDCODE_CONSTRAINT_VALUES, NO);
+    mpc.set( CG_USE_VARIABLE_WEIGHTING_MATRIX, NO);
+
+    if (mpc.exportCode( "." ) != SUCCESSFUL_RETURN)
```

Comment:
maybe export code in `../solver` directory. Please give it a try before we merge.

Reply:
Good point! We have used it in the way that we have just renamed the directory to solver and removed the old one (just renamed it to solver_made_from_matlab). 

I was unable to specify  mpc.exportCode( "../solver" ).

However it could change the local CMakeLists.txt in the folder to set the target and executable to ../solver/nmpc_solver and it seems to work just fine. Then one need to go to the solver directory and then run the executable file. I get a warning from when running "cmake .." though since i think it is not adviced to place the executable file in another directory. 

Do you want be to go with this solution?